src/pages/messages/components/MessageList.tsx

```diff
@@ -73,52 +117,85 @@ export const MessageItem = ({ text, date, username, backgroundColor, onDelete, i
         bgcolor: backgroundColor,
         p: '0.5rem',
         borderRadius: '4px',
-        cursor: onDelete && !expanded ? 'pointer' : 'auto',
         display: 'flex',
         flexDirection: 'column',
-      }}
-      onClick={() => setExpanded(true)}>
-      <Typography sx={{ display: 'flex', flexWrap: 'wrap', justifyContent: 'space-between' }}>
-        <span>
-          {senderQuery.isPending ? (
-            <Skeleton sx={{ width: '8rem' }} />
-          ) : (
-            <b data-testid={dataTestId.registrationLandingPage.tasksPanel.messageSender}>
-              {senderName ? senderName : <i>{t('common.unknown')}</i>}
-            </b>
-          )}
-        </span>
-        <span data-testid={dataTestId.registrationLandingPage.tasksPanel.messageTimestamp}>{toDateString(date)}</span>
-      </Typography>
+      }}>
+      <Box sx={{ display: 'grid', gridTemplateColumns: '1fr auto', alignItems: 'center' }}>
+        <Typography sx={{ display: 'flex', flexWrap: 'wrap', justifyContent: 'space-between' }}>
+          <Tooltip title={senderName ? senderName : t('common.unknown')}>
+            <span style={{ marginRight: '0.5rem' }}>
+              {senderQuery.isPending ? (
+                <Skeleton sx={{ width: '8rem' }} />
+              ) : (
+                <b data-testid={dataTestId.registrationLandingPage.tasksPanel.messageSender}>
+                  {senderName ? truncateName(senderName, 20) : <i>{t('common.unknown')}</i>}
+                </b>
+              )}
+            </span>
+          </Tooltip>
+          <span data-testid={dataTestId.registrationLandingPage.tasksPanel.messageTimestamp}>{toDateString(date)}</span>
+        </Typography>
+        {canDeleteMessage && onDelete && (
+          <section>
+            <IconButton
+              aria-controls={open ? 'basic-menu' : undefined}
+              aria-haspopup="true"
+              aria-expanded={open ? 'true' : undefined}
```

Comment:
VI skulle vært mye flinkere til sånt der det trengs ja 😬 Bra! 👏 
Problemet mitt er at jeg lener meg alt for mye på `axe`, men den kan ikke vite at knappet åpner en popup, så egentlig QA som må passe på dette tror jeg. Har heller ikke full oversikt over hva som skal inn for alle situasjoner, og hva som ev. gjør mer skade enn nytte.. :/